app/SupportDetection.jsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+
+import './styles/support-detection.scss';
+
+const { detect } = require('detect-browser');
+
+const browser = detect();
+
+const SupportedBrowser = {
+    'Mac OS': {
+        safari: 808,
+        firefox: 2600,
+        chrome: 2901547,
+        opera: 1601190,
+        yandexbrowser: 14122120,
+    },
+    'Windows 10':{
+        ie: 0,
+        firefox: 0,
+        chrome: 0,
+        opera: 0,
+        yandexbrowser: 0,
+    },
+    'Windows 8':{
+        ie: 1000,
+        firefox: 2700,
+        chrome: 3601985,
+        opera: 2601656,
+        yandexbrowser: 14122125,
+    },
+    'Windows 7':{
+        ie: 1100,
+        firefox: 2700,
+        chrome: 3601985,
+        opera: 2601656,
+        yandexbrowser: 0,
+    },
+};
+
+
+class SupportDetection extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            unknown: false,
+            hasError: false,
+        };
+    }
+
+    componentWillMount() {
+        if (browser) {
+            const version = browser.version.split('.').join('');
+            /**
+                safari, chrome, firefox, opera, yandexbrowser, ie, edge
+            */
+            const name = browser.name;
+            /**
+                Mac OS, IOS, Android OS, Windows 7, Windows 8, Windows 10,
+            */
+            const os = browser.os;
+
+            if (SupportedBrowser[os] && SupportedBrowser[os][name]) {
+                this.setState({
+                    hasError: (version < SupportedBrowser[os][name]),
+                });
+            } else {
+                this.setState({ unknown: true });
+            }
+        }
+    }
+
+    componentDidCatch(error, info) {
+        this.setState({ hasError: true });
+        console.error(error, info);
+    }
+
+    render() {
+        const NotGuaranteed = (
+            <div className="cq-not-guaranteed">
+                <h1>We are  NOT guaranteed</h1>
+                <p>Binary.com does not guaranteed your browser. Please update your browser.</p>
```

Comment:
Same here. Maybe we can get our copy writer to have a look for a better message.

Reply:
Do we need to translate that message? THere's no way of setting language tho

Reply:
use this:
<h1>Incompatible browser</h1>
<p>Binary.com does not recognise or support your current browser. Please *update your browser* immediately.</p>

Reply:
@ashkanx Any special design? for example for the button `update your browser`.

Reply:
Hi @shayanbinary , I sent you a link for the design we're using on smart-trader.